### ISSUE TITLE
[Machine Learning] Snorkel Python for Labeling Datasets Programmatically

### ISSUE THREAD

**Issue by charles721** [section-engineering-education/engineering-education#4189]
#### Proposed title of article
[Machine Learning]Snorkel Python for Labeling Datasets Programmatically
#### Proposed article introduction

Snorkel is a system for programmatically building and managing training datasets without manual labeling. Rather than hand-labeling thousands of data points by hand, Snorkel programmatically labels massive amounts of training data using labeling functions—rules, heuristics, and other custom complex operators.

Snorkel Flow automatically learns the different labeling functions’ accuracies, denoises and integrates them, and stores versioned packages and training data in Data Manager. Unlike with hand-labeled data, you create training data in Snorkel Flow using code, so you can audit, modify, or serve it almost instantly. Snorkel Flow makes it easy to share resources and training data, with others on the team.

Snorkel requires users to write rules that can label some portion of data. These rules are high-level rules that are written with the help of domain experts. In Snorkel terms, these rules are called Labeling Functions. These labeling functions are applied to all unlabeled data. The transformed data is sent to the core function of Snorkel known as Label Model which estimates the individual labeling function accuracy and correlations amongst them to assign labels to unlabelled data. The snorkel also provides slicing functions which are used to observe label model performance on slices of data

Finally, provides access to popular open-source model libraries that you can train on CPU- or GPU-based computing infrastructure. Snorkel Flow makes it easy to tune your models with automated hyperparameter search.

Snorkel has the following functions.
- Labeling Function : for labeling unlabelled data
- Transformation Function: for data augmentation
- Slicing Function: for data subset selection for important dataset
#### Key takeaways 
1. What is Snorkel Python and how to install it.
2. Key features of data labeling.
3. How to implement Labeling Function, Transformation Function, and Slicing Function.
4. Labelling a sample dataset.

#### Article quality
My article covers a unique and original project to explain the concepts of Snorkel Python. It's a practical guide that follows all the stages of preparing our dataset before labeling it with the snorkel functions.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)
